### PR TITLE
Use @Instance for mod lifecycle.

### PR DESCRIPTION
**Pull request by kylev**
This is janitorial work to bring things up to date with modern Forge
practice. There's no need for a factory, just annotate a public data
member and Forge promises to manage the mod as a singleton.

The factory method gets removed, too. We don't need it since Forge is
fine just calling the (now public) no-arg constructor.

### CODE REVIEW

File: src/main/java/appeng/util/Platform.java [hunk 1]
```diff
@@ -374,15 +374,15 @@ public static void openGUI( @Nonnull final EntityPlayer p, @Nullable final TileE
...
 			}
 			else if( tile == null || type.getType() == GuiHostType.ITEM )
 			{
-				p.openGui( AppEng.instance(), type.ordinal() << 4 | ( 1 << 3 ), p.getEntityWorld(), x, y, z );
+				p.openGui( AppEng.instance, type.ordinal() << 4 | ( 1 << 3 ), p.getEntityWorld(), x, y, z );
 			}
 			else
 			{
-				p.openGui( AppEng.instance(), type.ordinal() << 4 | ( side.ordinal() ), tile.getWorld(), x, y, z );
+				p.openGui( AppEng.instance, type.ordinal() << 4 | ( side.ordinal() ), tile.getWorld(), x, y, z );
```
Comment (orod-org) [inline src/main/java/appeng/util/Platform.java]: MAJOR A "NullPointerException" could be thrown; "side" is nullable here. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/appeng/core/AppEng.java [hunk 2]
```diff
@@ -92,8 +91,8 @@
 	// + net.minecraftforge.common.ForgeVersion.revisionVersion + '.' // revisionVersion
 	// + net.minecraftforge.common.ForgeVersion.buildVersion + ",)"; // buildVersion
 
-	@Nonnull
-	private static final AppEng INSTANCE = new AppEng();
+	@Instance( MOD_ID )
+	public static AppEng instance;
```
Comment (orod-org) [inline src/main/java/appeng/core/AppEng.java]: MINOR Make this "public static instance" field final [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1444)

**Comment by orod-org** [comment]
SonarQube analysis reported 30 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 13 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [Platform.java#L1409](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L1409): Refactor this method to not always return the same value. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3516)
1. ![CRITICAL][CRITICAL] [ItemSpatialStorageCell.java#L155](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/items/storage/ItemSpatialStorageCell.java#L155): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PathGridCache.java#L355](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/me/cache/PathGridCache.java#L355): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [Platform.java#L515](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L515): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [Platform.java#L1440](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L1440): The Cyclomatic Complexity of this method "extractItemsByRecipe" is 21 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![CRITICAL][CRITICAL] [Platform.java#L1458](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L1458): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [Platform.java#L1474](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L1474): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [PathGridCache.java#L317](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/me/cache/PathGridCache.java#L317): A "HashSet<IPathItem>" cannot contain a "IGridNode" [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2175)
1. ![MAJOR][MAJOR] [StorageHelper.java#L183](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/spatial/StorageHelper.java#L183): Method has 11 parameters, which is greater than 7 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Platform.java#L515](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/f1d18ff103b46f2788d471ac373668eae01a605b/src/main/java/appeng/util/Platform.java#L515): Merge this if statement with the enclosing one. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by yueh** [comment]
Relying on a public field is in general bad practise. E.g. when having to rename it for various reasons like fml suddenly injecting it at a different point of time. Also with FML2 in development and potential breaking, it is not really worth to change a working approach for the sake of changing it and then redoing it anyway.

**Pull request closed without merging**